scripts/metric_reporter/averages_reporter.py

```diff
@@ -0,0 +1,174 @@
+            "Success Rate 30": self.success_rate_30,
+            "Run Time 30": self.run_time_30,
+            "Execution Time 30": self.execution_time_30,
+            "Job Time 30": self.job_time_30,
+            "Start Date 60": self.start_date_60,
+            "End Date 60": self.stop_date_60,
+            "Suite Count 60": self.suite_count_60,
+            "Success Rate 60": self.success_rate_60,
+            "Run Time 60": self.run_time_60,
+            "Execution Time 60": self.execution_time_60,
+            "Job Time 60": self.job_time_60,
+            "Start Date 90": self.start_date_90,
+            "End Date 90": self.stop_date_90,
+            "Suite Count 90": self.suite_count_90,
+            "Success Rate 90": self.success_rate_90,
+            "Run Time 90": self.run_time_90,
+            "Execution Time 90": self.execution_time_90,
+            "Job Time 90": self.job_time_90,
+        }
+
+
+class AveragesReporter(BaseReporter):
+    """Handles the reporting of test suite results from CircleCI metadata and JUnit XML Reports."""
+
+    def __init__(
+        self,
+        repository: str,
+        workflow: str,
+        test_suite: str,
+        suite_results: Sequence[SuiteReporterResult],
+    ) -> None:
+        """Initialize the reporter with the test suite results.
+
+        Args:
+            repository (str): The repository associated to the test suite.
+            workflow (str): The workflow associated to the test suite.
+            test_suite (str): The test suite name.
+            suite_results (Sequence[SuiteReporterResult]): Test suite results.
+        """
+        super().__init__()
+        self.results = self._parse_results(repository, workflow, test_suite, suite_results)
+
+    def _parse_results(
+        self,
+        repository: str,
+        workflow: str,
+        test_suite: str,
+        suite_results: Sequence[SuiteReporterResult],
+    ) -> list[AveragesReporterResult]:
+        results: list[AveragesReporterResult] = []
+        if not suite_results:
+            return results
+
+        # Assumes that suite_results is sorted
+        first_date: str | None = suite_results[0].date
+        last_date: str | None = suite_results[-1].date
+        if not first_date or not last_date:
+            return results
+
+        first_datetime: datetime = datetime.strptime(first_date, DATE_FORMAT)
+        last_datetime: datetime = datetime.strptime(last_date, DATE_FORMAT)
+
+        while last_datetime >= first_datetime + timedelta(days=DAYS_30):
+            averages_dict = {
+                "repository": repository,
+                "workflow": workflow,
+                "test_suite": test_suite,
+                **self._calculate_averages(suite_results, last_datetime, DAYS_30),
+                **self._calculate_averages(suite_results, last_datetime, DAYS_60),
+                **self._calculate_averages(suite_results, last_datetime, DAYS_90),
+            }
```

Comment:
Should this have a type set?

Reply:
Sure! (see https://github.com/mozilla/ecosystem-test-scripts/pull/13/commits/8d0b244b0924bcd02918c2f649261ffdb076db0f)